### DISCUSSION TITLE
Bug: WebServer Route Any Method

### DISCUSSION THREAD

**Discussion by QDaniel** [nanoframework/Home#1560] [General]
### Library/API/IoT binding

nanoFramework.WebServer

### Visual Studio version

17.12.2

### Description

The WebServer only allows GET Requests if the Method Attribute not set

`[Route("test/any")]
    public void RouteAnyTest(WebServerEventArgs e)
 `

https://github.com/nanoframework/nanoFramework.WebServer/blob/57a1653346fd469f3d86b8f875e69356e8723e36/nanoFramework.WebServer/WebServer.cs#L693
`            if (method != (string.IsNullOrEmpty(route.Method) ? DefaultRouteMethod : route.Method))`

must be changed to 
`            if (method != (string.IsNullOrEmpty(route.Method) ? method : route.Method))`

### Expected behaviour

Allow all Methos if MethodAttribute not set

**Comment by Ellerbach** [comment]
Hi,
That's I think the normal behavior as if nothing is specified, then it's by default GET. Otherwise, you have to specify a method. It's working like this as well in ASP.NET as far as I can remember.
Moving this to discussion as it's not a but.
